asab/api/doc.py

```diff
@@ -38,55 +41,88 @@ def build_swagger_specs(self):
 				try:
 					adddict = yaml.load(docstr[i:], Loader=yaml.SafeLoader)
 				except yaml.YAMLError as e:
-						L.error("Failed to parse '{}' doc string {}".format(self.App.__class__.__name__, e))
+					L.error("Failed to parse '{}' doc string {}".format(self.App.__class__.__name__, e))
 			else:
 				description = docstr
 		else:
 			description = ""
 
-
 		specs = {
 			"openapi": "3.0.1",
 			"info": {
 				"title": "{}".format(self.App.__class__.__name__),
 				"description": description,
 				"contact": {
 					"name": "ASAB microservice",
-					"url": "http://www.github.com/teskalabs/asab",
+					"url": "https://www.github.com/teskalabs/asab",
 				},
 				"version": "1.0.0"
 			},
-			"servers": [
-				{"url": "../../"}  # Base path relative to openapi endpoint
-			],
-			"paths": {
+			"servers": [{"url": "../../"}],  # Base path relative to openapi endpoint
+			"paths": {},
+
+			# Authorization
+			"components": {
+				"securitySchemes": {
+					"oAuthSample": {
+						"type": "oauth2",
+						"description": "This API uses OAuth 2, this window allows [links](https://teskalabs.com)",
+						"flows": {
+							"authorizationCode": {
+								"authorizationUrl": "https://auth.test.loc/seacat/api/openidconnect/authorize",
+								"tokenUrl": "https://auth.test.loc/seacat/api/openidconnect/token",
+								"scopes": {
+									"write_stuff": "modify stuff :D",
+									"read_stuff": "read stuff :DD",
+									"openid": "openid",
+								}
+							}
+						}
+					}
+				},
 			},
 		}
 
+		# Show what server/docker container you are on, and it's IP
+		for server in self.WebContainer.Addresses:
```

Comment:
There is a better way - you need to publish only one address.
Check advertisement to zookeeper how it is done … it is the same task

Reply:
Think I fixed it.